tests/ui/xor_used_as_pow.rs

```diff
@@ -0,0 +1,33 @@
+// run-rustfix
+#![warn(clippy::xor_used_as_pow)]
+
+// Should not be linted
+#[allow(dead_code)]
+enum E {
+    First = 1 ^ 8,
+    Second = 2 ^ 8,
+    Third = 3 ^ 8,
+    Tenth = 10 ^ 8,
+}
+
+fn main() {
+    // These should succeed:
+    let _ = 9 ^ 3; // lhs other than 2 or 10
+    let _ = 0x02 ^ 6; // lhs not decimal
+    let _ = 2 ^ 0x10; // rhs hexadecimal
+    let _ = 10 ^ 0b0101; // rhs binary
+    let _ = 2 ^ 0o1; // rhs octal
+    let _ = 10 ^ -18; // negative rhs
+
+    // These should fail
+    let _ = 2 ^ 3;
+    let _ = 10 ^ 4;
+    let _ = 2 ^ 32;
+    let _ = 2 ^ 0;
+    let _ = 10 ^ 0;
```

Comment:
It looks like lints are emitted overlapping with `identity_op`. Should `xor_used_as_pow` be emitted in these pattern?

The test in `.fixed` seems fail due to `identity_op`. https://github.com/rust-lang/rust-clippy/pull/6182#issuecomment-714518744.

Reply:
I think that both should be emitted, actually. One reports that it's a useless operation, and the other asks if they meant to do exponentiation. It's possible they could be combined, though.

Reply:
Then the test can work around this by `#![allow(clippy::identity_op)]`.

Reply:
I stopped linting for `identity_op` and the test is still failing. It seems to be because this new lint doesn't always offer suggestions, and so the fixed file still has code that is linted. Is there a way around that?

Reply:
If I'm not mistaken in general we try to not give multiple messages that can contradict themselves and confuse the user, so only one lint should trigger in this case.

I think (like @giraffate suggests) that this lint should not trigger then when the RHS is `0`, and be covered only by `identity_op`. The error message will already give a clear clue that the operation is not exponentiation.

For the rustfix problem, you should split the tests in two files, `xor_used_as_pow.rs` and `xor_used_as_pow_fixable.rs`, and only add the `run-rustfix` header command to the fixable one.